### PR TITLE
autoconf: update to 2.71.

### PR DESCRIPTION
**Pull request by motorto**
[ci skip]

#### Testing the changes
- I tested the changes in this PR: **YES**

#### Local build testing
- I built this PR locally for my native architecture, x86_64-glibc
- I built this PR locally for these architectures (if supported. mark crossbuilds):
  - armv6l

I have also update autoconf-archive in this pr.

Because this is a critical package I will make sure that all package that depends on autoconf and automake still compile.

There was some changes that might need to add [gtk-doc on hostmakedepends](https://www.mail-archive.com/bug-autoconf@gnu.org/msg04872.html). Separate PRs for the packages will be created and I will reference here.

Compiled all the packages that depend on autoconf and automake `grep -ri autconf` (Yes, I know it has some false negatives).  Thanks @paper42  for that.

Packages failing that were failing already (will not fix)

- ctags (patch fails to apply ...) #38597 
- davfs2 (fixed in dd5000ae0e651956977c9280142fd45678df6916)
- gamin (patch fails to apply, fixed in 643353ee640c320645c7946adb8c9862dcbdef61) 
- gsoap (fail to fetch needs to be updated)
- kea -  fails with sphinx.errors.SphinxWarning: Invalid configuration value found: 'language = None'. Update your configuration to a valid langauge code. Falling back to 'en' (English). fixed in fee715adc214d07b022f25881b8840bc8494fe69
- libcryptui (fixed in 26e2d9fbdc2ff2d644f141cce4d48d22921694b0, also fixed for autoconf 2.71 in this PR)
- libtar (patchs fail to apply)
- mod_wsgi
- pcsxr (fixed in 580473cab7f407c005080c4accf98316e5a45e60)
- pdsh ( tests fail)
- pgadmin3 (package removed in e5860d61e94d3e37c824378bdb4971dc54eaec16)
- rng-tools
- tcpflow
- transcode
- xnec2c (#38696)

Packages that the tests were failling already ~(added make_check=no)~

- clutter
- openldap
- rest
- watchman

Packages fixed on other PRS:

- bluebird-themes #38543 
- tpm2-tss #38567 
- lksctp-tools #38581 
- tigervnc #38647
- hplip #38649
- seahorse-nautilus #38650

Packages fixed in this PR:

- geany 
- hunspell
- iverilog
- libsasl - remove patch for old autoconf
- libsrtp
- ltrace
- lxsession
- pilot-link
- openjdk7-bootstrap
- gnome-latex
- libcryptui
- nfs4-acl-tools
- portaudio

Packages that need to tested by someone else (don't build on x86_64{musl,glibc})

- librtas
- libump-git
- libvpd
- lsvpd
- powerpc-utils
- sun7i-kernel
- yaboot

Packages that failed on @paper42  build but compile fine (might fail because of -j20)

- firefox - I am not trying to compile this one on my system sorry :)
- mbox - compiles fine on my system
- mozjs78 - compiles fine on my system
- mozjs91 -  compiles fine on my system
- opensc -  compiles fine on my system
- thunderbird
- tracebox - compiles fine on my system
- vlc - compiles fine on my system
- xmoto - compiles fine on my system
- gloobus-preview

### CODE REVIEW

**Comment by paper42** [comment]
this is a core package, briefly testing it is not enough

**Comment by motorto** [comment]
@paper42  any way to force xbps-src to use this autoconf instead of the ones in the repos ?

**Comment by paper42** [comment]
> @paper42  any way to force xbps-src to use this autoconf instead of the ones in the repos ?

make sure this commit is applied in the current branch when you are building something

**Comment by motorto** [comment]
> > @paper42  any way to force xbps-src to use this autoconf instead of the ones in the repos ?
> 
> make sure this commit is applied in the current branch when you are building something

sure, my bad I was being dumb. 

I will try to build base-system with -N option with this autoconf update and I will report back

**Comment by motorto** [comment]
Pinging the maintainers to see if they can help (updating to last version might do the trick)
@thypon - rng-tools
@evandeaubl  tigervnc

**Comment by paper42** [comment]
`ltrace: update to 0.7.3.20160924.`  commit message is wrong, it's not an update

File: srcpkgs/ltrace/template [hunk 1]
```diff
@@ -25,5 +25,7 @@ pre_configure() {
 		sed -i 's/#ifndef __mips__/#if 0/' proc.c
 	fi
 
-	autoreconf -fi
+	mkdir -p config 
+	autoreconf -fiv
```
Comment (paper42) [inline srcpkgs/ltrace/template:29]: remove -v

File: srcpkgs/geany/template [hunk 2]
```diff
@@ -17,10 +17,6 @@ distfiles="https://download.${pkgname}.org/${pkgname}-${version}.tar.bz2"
 checksum=abff176e4d48bea35ee53037c49c82f90b6d4c23e69aed6e4a5ca8ccd3aad546
 python_version=3
 
-pre_configure() {
-	autoreconf -fi
-}
```
Comment (paper42) [inline srcpkgs/geany/template:22]: if we are removing autoreconf, we can probably remove some hostmakedepends

File: srcpkgs/gloobus-preview/template [hunk 3]
```diff
@@ -23,5 +23,6 @@ if [ -z "$CROSS_BUILD" ]; then
 fi
 
 pre_configure() {
-	autoreconf -fi
+	sed -i '/ACX_PTHREAD()/d' configure.ac
```
Comment (paper42) [inline srcpkgs/gloobus-preview/template:26]: use vsed and add a description why this is necessary

File: srcpkgs/gloobus-preview/template [hunk 4]
```diff
@@ -23,5 +23,6 @@ if [ -z "$CROSS_BUILD" ]; then
...
+	autoreconf -fiv
```
Comment (paper42) [inline srcpkgs/gloobus-preview/template:27]: remove -v

File: srcpkgs/hplip/template [hunk 5]
```diff
@@ -27,7 +27,7 @@ make_dirs="/var/lib/hp 0755 root root"
 hostmakedepends="pkg-config automake libtool python3"
 makedepends="openssl-devel python3-devel libxml2-python3 cups-devel sane-devel
  ghostscript-devel net-snmp-devel libusb-devel libjpeg-turbo-devel dbus-devel
- avahi-libs-devel"
+ avahi-libs-devel zlib-devel"
```
Comment (paper42) [inline srcpkgs/hplip/template:30]: this is also broken without the autoconf update, drop it from here and create a new separate PR for this fix

File: srcpkgs/libsrtp/template [hunk 6]
```diff
@@ -19,10 +19,6 @@ post_install() {
 	vlicense LICENSE
 }
 
-pre_configure() {
-	autoreconf -fi
-}
```
Comment (paper42) [inline srcpkgs/libsrtp/template:24]: automake can be probably dropped from hostmakedepends

File: srcpkgs/lxsession/template [hunk 7]
```diff
@@ -17,5 +17,4 @@ replaces="lxpolkit>=0"
 
 pre_build() {
 	rm *.stamp
-	autoreconf -fi
```
Comment (paper42) [inline srcpkgs/lxsession/template:20]: autoconf/automake/libtool can probably be dropped from hostmakedepends

File: srcpkgs/openldap/template [hunk 8]
```diff
@@ -31,7 +31,6 @@ pre_configure() {
 	sed -i 's|#define LDAPI_SOCK LDAP_RUNDIR LDAP_DIRSEP "run" LDAP_DIRSEP "ldapi"|#define LDAPI_SOCK LDAP_DIRSEP "run" LDAP_DIRSEP "openldap" LDAP_DIRSEP "ldapi"|' include/ldap_defaults.h
 	sed -i 's|%LOCALSTATEDIR%/run|/run/openldap|' servers/slapd/slapd.{conf,ldif}
 	sed -i 's|-$(MKDIR) $(DESTDIR)$(localstatedir)/run|-$(MKDIR) $(DESTDIR)/run/openldap|' servers/slapd/Makefile.in
-	aclocal; autoconf; autoheader; libtoolize -f
```
Comment (paper42) [inline srcpkgs/openldap/template:34]: some hostmakedepends can be dropped

File: srcpkgs/seahorse-nautilus/patches/new-gnupg.patch [hunk 9]
```diff
@@ -7,7 +7,7 @@ Index: seahorse-nautilus-3.10.1/configure.ac
  
  if test	"$DO_CHECK" = "yes"; then
 -	accepted_versions="1.2 1.4 2.0"
-+	accepted_versions="1.2 1.4 2.0 2.1 2.2"
++	accepted_versions="1.2 1.4 2.0 2.1 2.2 2.3"
```
Comment (paper42) [inline srcpkgs/seahorse-nautilus/patches/new-gnupg.patch:10]: move to a separate PR, this is not relevant for the autoconf update

File: srcpkgs/tigervnc/template [hunk 10]
```diff
@@ -6,7 +6,7 @@ _xorg_version=1.20.0
 build_style=cmake
 hostmakedepends="automake gettext xorg-util-macros font-util pkg-config libtool xtrans"
 makedepends="fltk-devel zlib-devel libXtst-devel libjpeg-turbo-devel pixman-devel
- xorgproto libxkbfile-devel libXrandr-devel libXfont2-devel gnutls-devel libdrm-devel"
+ xorgproto libxkbfile-devel libXrandr-devel libXfont2-devel gnutls-devel libdrm-devel libxshmfence-devel"
```
Comment (paper42) [inline srcpkgs/tigervnc/template:9]: separate PR please

File: srcpkgs/watchman/patches/fix_autogen.patch [hunk 11]
```diff
@@ -0,0 +1,13 @@
+From: https://github.com/facebook/watchman/issues/937#issuecomment-904717182
```
Comment (paper42) [inline srcpkgs/watchman/patches/fix_autogen.patch:1]: build still fails with even without this autoconf update:
```
pywatchman/bser.c:31:10: fatal error: Python.h: No such file or directory                                                                         
   31 | #include <Python.h>                                                                                                                       
      |          ^~~~~~~~~~                                                                                                                       
compilation terminated.
```

File: srcpkgs/gloobus-preview/template (hunk 4 above)
Comment (motorto) [reply srcpkgs/gloobus-preview/template]: Weirdly enough now it is compiling even without the sed command... 

`./xbps-src clean && ./xbps-src pkg ` gives me a clean build state right ?

File: srcpkgs/watchman/patches/fix_autogen.patch (hunk 11 above)
Comment (motorto) [reply srcpkgs/watchman/patches/fix_autogen.patch:1]: In the autoconf 2.7.1 branch it builds fine: https://paste.rs/Gwu

I didn't had time to try with autoconf 2.6.9

Comment (motorto) [reply srcpkgs/watchman/patches/fix_autogen.patch:1]: With autoconf 2.6.9 also builds: https://paste.rs/wGs

Comment (paper42) [reply srcpkgs/watchman/patches/fix_autogen.patch:1]: Hmm, I must have had a dirty masterdir, I can't reproduce this failure anymore.

File: srcpkgs/gloobus-preview/template (hunk 4 above)
Comment (paper42) [reply srcpkgs/gloobus-preview/template]: mostly, clean only cleans directories in /builddir and /destdir/. If one of the previous builds touched the system outside of these directories (and you were not using -t), it would still be broken. `zap` and `binary-bootstrap` recreate the masterdir from scratch, so that definitely gives you a clean state. `clean` should be good enough for all packages.

**Comment by paper42** [comment]
farstream needs gtk-doc added to hostmakedepends

File: srcpkgs/gloobus-preview/template (hunk 4 above)
Comment (motorto) [reply srcpkgs/gloobus-preview/template]: Tried building now this package after a zap 

could you confirm that this package builds without the sed ?

**Comment by paper42** [comment]
@motorto please resolve conflicts
